### PR TITLE
Switch to SQLAlchemy

### PR DESCRIPTION
**Pull request by gentlecat**
**Proper diff: https://github.com/metabrainz/listenbrainz-server/compare/cleanup...nastybad.**

### CODE REVIEW

**Comment by gentlecat** [comment]
Didn't get a chance to test that yet. Some old code needs to be updated as well (like database initialization functions in *manage.py*).

File: config.py.sample [hunk 1]
```diff
@@ -2,23 +2,24 @@ DEBUG = False  # set to False in production mode
...
-# See http://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING
-# Note: To have no password, use "password=" in the string below.
-PG_CONNECT = "dbname=listenbrainz user=listenbrainz password= host=localhost port=5432"
+# Primary database
+SQLALCHEMY_DATABASE_URI = "postgresql://listenbrainz@/listenbrainz"
 
-# This connection string will be used for testing
-PG_CONNECT_TEST = "dbname=lb_test user=lb_test password= host=localhost port=5432"
+# Database for testing
+TEST_SQLALCHEMY_DATABASE_URI = "postgresql://lb_test:lb_test@localhost:5432/lb_test"
```
Comment (alastair) [inline config.py.sample]: this has a password set, was that intentional?

File: db/__init__.py [hunk 2]
```diff
@@ -13,38 +6,21 @@
...
-    _connection.close()
+    global db
+    db.init_app(app)
 
 
 def run_sql_script(sql_file_path):
-    with _connection.cursor() as cursor:
-        with open(sql_file_path) as sql:
-            cursor.execute(sql.read())
+    global db
```
Comment (alastair) [inline db/__init__.py]: remember, you don't need global if you're not setting the variable

File: db/user.py [hunk 3]
```diff
@@ -1,44 +1,40 @@
...
-                               VALUES (%s, %s)
-                            RETURNING id""",
-                       (musicbrainz_id, str(uuid.uuid4())))
-        commit()
-        new_id = cursor.fetchone()[0]
-        return new_id
+    # TODO(roman): Do we need to make sure that musicbrainz_id is case insensitive?
+    result = db.session.execute(
+        'INSERT INTO "user" (musicbrainz_id, auth_token) VALUES (:mb_id, :token) '
+        'RETURNING id',
```
Comment (alastair) [inline db/user.py]: Please format this like the query was (keywords on new lines, right aligned)

Comment (alastair) [inline db/user.py]: Use `"""` so that you don't need to quote each line

File: config.py.sample (hunk 1 above)
Comment (gentlecat) [inline config.py.sample]: I just forgot to change it.

File: db/__init__.py [hunk 4]
```diff
@@ -13,38 +6,19 @@
 # Be careful when importing `_connection` before init_connection function is
```
Comment (alastair) [inline db/__init__.py]: This comment is out of date now

**Comment by alastair** [comment]
I'm happy with this now. we should merge and deploy.

File: db/__init__.py (hunk 4 above)
Comment (gentlecat) [inline db/__init__.py]: Removed.

**Pull request merged**